File: java/SpringBoot/cloud_function/sample/src/main/java/com/example/Application.java
package com.example;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import java.util.function.Function;

@SpringBootApplication
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Bean
    public Function<ItemInput, Item> sample() {
        return input -> new Item("item-" + input.value(), input.value());
    }

    public record ItemInput(int value) {}
    public record Item(String id, int value) {}
}

File: kotlin/spring/webflux/websocket-sample/src/main/kotlin/sample/Application.kt
package sample

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.HandlerMapping
import org.springframework.web.reactive.config.EnableWebFlux
import org.springframework.web.reactive.config.ResourceHandlerRegistry
import org.springframework.web.reactive.config.WebFluxConfigurer
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping
import org.springframework.web.reactive.socket.server.support.WebSocketHandlerAdapter
import sample.controller.SampleHandler

@SpringBootApplication
@EnableWebFlux
@Configuration
class Application : WebFluxConfigurer {
    override fun addResourceHandlers(registry: ResourceHandlerRegistry?) {
        registry?.let {
            it.addResourceHandler("/static/**")
                    .addResourceLocations("classpath:static/")
        }
    }

    @Bean
    fun webSocketMapping(): HandlerMapping {
        val mapping = SimpleUrlHandlerMapping()

        mapping.urlMap = mapOf("/sample" to SampleHandler())

        return mapping
    }

    @Bean
    fun handlerAdapter() = WebSocketHandlerAdapter()
}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}
